File: erpnext/accounts/report/other_deposit/other_deposit.js
// Copyright (c) 2026, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Other Deposit"] = {
	 filters: [
        {
            fieldname: "company",
            label: __("Company"),
            fieldtype: "Link",
            options: "Company",
            default: frappe.defaults.get_user_default("Company"),
        },
        {
            fieldname: "from_date",
            label: __("From Date"),
            fieldtype: "Date",
            default: frappe.datetime.month_start(),
        },
        {
            fieldname: "to_date",
            label: __("To Date"),
            fieldtype: "Date",
            default: frappe.datetime.month_end(),
        },
        {
            fieldname: "account",
            label: __("Account"),
            fieldtype: "Link",
            options: "Account",
            reqd: 1,
            get_query: function() {
                return {
                    filters: {
                        company: frappe.query_report.get_filter_value("company"),
                        is_group: 0
                    }
                };
            }
        },
       {
            fieldname: "party_type",
            label: __("Party Type"),
            fieldtype: "Autocomplete",
            options: Object.keys(frappe.boot.party_account_types),
            on_change: function () {
                frappe.query_report.set_filter_value("party", "");
            },
        },
        {
            fieldname: "party",
            label: __("Party"),
            fieldtype: "MultiSelectList",
            get_data: function (txt) {
                if (!frappe.query_report.filters) return;

                let party_type = frappe.query_report.get_filter_value("party_type");
                if (!party_type) return;

                return frappe.db.get_link_options(party_type, txt);
            },
            on_change: function () {
                var party_type = frappe.query_report.get_filter_value("party_type");
                var parties = frappe.query_report.get_filter_value("party");

                if (!party_type || parties.length === 0 || parties.length > 1) {
                    frappe.query_report.set_filter_value("party_name", "");
                    frappe.query_report.set_filter_value("tax_id", "");
                    return;
                } else {
                    var party = parties[0];
                    var fieldname = erpnext.utils.get_party_name(party_type) || "name";
                    frappe.db.get_value(party_type, party, fieldname, function (value) {
                        frappe.query_report.set_filter_value("party_name", value[fieldname]);
                    });

                    if (party_type === "Customer" || party_type === "Supplier") {
                        frappe.db.get_value(party_type, party, "tax_id", function (value) {
                            frappe.query_report.set_filter_value("tax_id", value["tax_id"]);
                        });
                    }
                }
            },
        },
    ],
    formatter: function(value, row, column, data, default_formatter) {

        value = default_formatter(
            value,
            row,
            column,
            data
        );

        if (
            column.fieldname === "total_outstanding" &&
            data &&
            flt(data.total_outstanding) < 0
        ) {
            value =
                '<span style="color:red;font-weight:bold;">' +
                value +
                '</span>';
        }
        return value;
    }
};

File: erpnext/accounts/report/other_deposit/other_deposit.py

import frappe


def execute(filters=None):
    filters = filters or {}

    columns = [
     
        {
            "label": "Party Type",
            "fieldname": "party_type",
            "fieldtype": "Data",
            "width": 120,
        },
        {
            "label": "Party",
            "fieldname": "party",
            "fieldtype": "Dynamic Link",
            "options": "party_type",
            "width": 200,
        },
        {
            "label": "Posting Date",
            "fieldname": "posting_date",
            "fieldtype": "Date",
            "width": 80,
        },
        {
            "label": "Voucher Type",
            "fieldname": "voucher_type",
            "fieldtype": "Data",
            "width": 120,
        },
        {
            "label": "Voucher No",
            "fieldname": "voucher_no",
            "fieldtype": "Dynamic Link",
            "options": "voucher_type",
            "width": 180,
        },

        
        {
            "label": "Opening Balance",
            "fieldname": "opening_balance",
            "fieldtype": "Currency",
            "width": 180,
        },
        {
            "label": "FY Balance",
            "fieldname": "fy_balance",
            "fieldtype": "Currency",
            "width": 180,
        },
       
  
        {
            "label": "Total Outstanding",
            "fieldname": "total_outstanding",
            "fieldtype": "Currency",
            "width": 120,
        },
    ]

    conditions = [
  
    ]

    values = {}

    if filters.get("company"):
        conditions.append("gl.company = %(company)s")
        values["company"] = filters["company"]

    if filters.get("from_date"):
        conditions.append("gl.posting_date >= %(from_date)s")
        values["from_date"] = filters["from_date"]

    if filters.get("to_date"):
        conditions.append("gl.posting_date <= %(to_date)s")
        values["to_date"] = filters["to_date"]

    if filters.get("account"):
        conditions.append("gl.account = %(account)s")
        values["account"] = filters["account"]

    if filters.get("party_type"):
        conditions.append("gl.party_type = %(party_type)s")
        values["party_type"] = filters["party_type"]

    if filters.get("party"):
        conditions.append("gl.party = %(party)s")
        values["party"] = filters["party"]

    data = frappe.db.sql(
		   f"""
		   SELECT
                gl.name,
                gl.posting_date AS posting_date,
                gl.account AS account,
                gl.party_type AS party_type,
                gl.party AS party,
                gl.voucher_type as voucher_type,
                gl.voucher_no as voucher_no,
                CASE
                    WHEN gl.is_opening = 'Yes' THEN gl.credit - gl.debit
                    ELSE 0
                END AS opening_balance,
                gl.credit as fy_balance,
                gl.credit-gl.debit as total_outstanding
		   FROM `tabGL Entry` gl
		   WHERE {" AND ".join(conditions)}
		   ORDER BY
			   gl.posting_date
		   """,
		   values,
		   as_dict=True,
	   )

    return columns, data
